pkg/deploy/controller/lifecycle/execnewpod/controller.go

```diff
@@ -0,0 +1,95 @@
+package execnewpod
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+// ExecNewPodController keeps an ExecNewPodAction pod's status in sync with
+// its associated deployment.
+//
+// Use ExecNewPodControllerControllerFactory to create this controller.
+type ExecNewPodController struct {
+	// deploymentClient provides access to deployments.
+	deploymentClient deploymentClient
+}
+
+// Handle syncs the action pod's status with its associated deployment.
+func (c *ExecNewPodController) Handle(pod *kapi.Pod) error {
```

Comment:
Why isn't this run by the deployment process itself?  Why does it need a controller?